### ISSUE TITLE
成為python數據分析達人的第一課_note

### ISSUE THREAD

**Issue by pspiskinghuayuworld** [pspiskinghuayuworld/learn_note#2]
## 互動(interact )
```py
from ipywidgets import interact
```
* **interact 的用法：**`interact (函數,變數=起始值)`
* 設定線的粗細： `plt.plot ( 橫軸, 縱軸, lw = 粗細數字)`
* 畫圖時間設定指令：`plt.show()`
* 畫點指令：`plt.scatter(X, Y)`
* `plt.scatter(x, y, s=100, c='b')`，s為大小，c為顏色
* `interact_manual` 和 `interact` 的差別是 `interact_manual` 要按按鈕才會改變產生出的東西

## Pandas
```python
import pandas as pd
```
* 讀 csv 的方法： `pd.read_csv("檔名")`
* `df = pd.read_csv("")` df 為 dataframe 縮寫
* 看 dataframe 前五行資料 ：`dataframe 名稱.head ( )`
* 取 dataframe 單項資料：`dataframe名稱[項目名稱]` or `dataframe名稱.項目名稱`
* dataframe單項資料轉 array ：`dataframe名稱[項目名稱].values` or `dataframe名稱.項目名稱.values`
* array 取平均值：`array名稱.mean( )`，array 取標準差： `array名稱.std( )`
* 畫單項資料折線圖：` dataframe 名稱.項目名稱.plot( )`
* 畫單項資料直方圖：` dataframe名稱.項目名稱.hist ( bins = 區分區間)`
* `DataFrame.describe()` 描述統計
* 合併dataframe ：`新dataframe = pd.concat(dataframe的list, axis = (0或1))` axis 控制合併的方向， row 為 0， column 為1
* 計算次數： `dataframe 名稱.count()`
* 依項目集合值：`dataframe 名稱.groupby()`
* 計算與前期的差值：`dataframe 名稱.diff( )`
## 預測
```py
from sklearn.linear_model import LinearRegression
```
* 開線性回歸機：`機器的變數名稱 = LinearRegression()`
* 改變 array 形狀：`array的變數名.reshape(新的形狀)`
* 開始讓線性回歸機學習：`機器的變數名稱.fit(輸入資料, 正確答案)`
* 開始預測：`機器的變數名稱.predict(輸入資料)`
* `x=  (0, 5, 50)` 代表 0 到 5 之間，取 50 個點
* 互動的時候要打`plt.show()`
* 分割 Training 和 Testing Data ：`輸入的訓練資料, 輸入的測試資料, 輸出的訓練資料, 輸出的測試資料 = train_test_split(x, y, test_size=分割給test的比例, random_state=隨便給一個數字)`
```py
from sklearn.model_selection import train_test_split
```
* 另外一種改變 array 形狀：`array 的變數名.shape = 新的形狀`
* `enumerate()`把元素個數與名稱算出來
## 機器學習入門
* 函數學習的兩大方法
1.監督式
2.非監督式
* pattern 人工分類： code 不須新增
1.`array([ [x,f(x)],[x,f(x)],[x,f(x)],.....])`-----> x--->f(x) series  資料結構
2.`plt.scatter(x,f(x),c=[1,2,2,3,....])`------> c:分類
* SVM 支持向量機
```py
from sklearn.svm import SVC
```
* 特徵：
1.n(列)xm(特徵)資料，以 series 一串為單位
2.將 m 種特徵分類，目前分為 [0,1,2] 三類
3.分類機` SVC.fit` 、 `SVC.predict `
4.視覺判斷 `plt.scatter`
*
